corehq/apps/data_dictionary/management/commands/populate_case_prop_groups.py

```diff
@@ -31,3 +33,17 @@ def populate_case_prop_groups(domain):
         )
         case_prop.group_obj = group
         case_prop.save()
+
+
+def remove_out_of_sync_prop_and_groups():
+    # Remove properties have different group and group_obj fields
+    properties_out_of_sync = (CaseProperty.objects
+                            .filter(group_obj__isnull=False)
+                            .filter(~Q(group_obj__name=F('group'))))
+    print("Remove out of sync groups for {} properties".format(len(properties_out_of_sync)))
+    properties_out_of_sync.update(group_obj=None)
```

Comment:
Should not this delete the group_obj as well? or they would get removed in the next query?

Reply:
The next query removes the groups with no properties, we can not delete the group here. Group might have other properties that reference it.